src/index.js

```diff
@@ -1,3 +1,423 @@
+
+   /**
+    * Create the instance variable that stores registered listeners if it does not
+    * already exist.
+    * @private
+    */
+   _ensureListenersHashExists: function() {
+      if (!this._eventListeners) {
+         this._eventListeners = {};
+      }
+   },
+
+   /**
+    * Register a listener function that will be called every time the specified event is
+    * emitted.
+    *
+    * Calls to `on` will de-duplicate listeners so that the same listener and context
+    * combination does not get invoked more than once for the same event. Also, calls
+    * to `on` override calls to {@link EventEmitterMixin#once} in that if there is still
+    * an event listener and context combination registered from a call to
+    * {@link EventEmitterMixin#once} and the same listener and context combination is
+    * passed to a call to `on`, that listener and context combination will **not** be
+    * removed after the first event.
+    *
+    * If the `listener` function (or the listener function and its associated `context`)
+    * was already registered using {@link EventEmitterMixin#on} or
+    * {@link EventEmitterMixin#once}, registering it again with `on` will have the
+    * following effect:
+    *
+    *    * `on`: if it was registered with `on`, nothing happens. There remains one
+    *    listener registered for the `eventNames` event(s).
+    *    * `once`: if it was registered with `once`, and the `eventName` event has not
+    *    been emitted yet, then that listener becomes an `on` listener, is executed each
+    *    time that the event is emitted, and is **not** removed after it has been called
+    *    once.
+    *
+    * @param eventNames {string} one or more names of the event(s) your listener will be
+    * invoked for, when emitted. Providing a string of space-separated names will bind
+    * the provided listener to each of the events listed.
+    * @param listener {function} the listener that will be called when this event is
+    * emitted
+    * @param [context] {object} the object that will be the `this` context for the
+    * `listener` function when it is executed. See the documentation on
+    * {@link EventEmitterMixin} for an explanation of when and how to use this parameter.
+    * @instance
+    * @returns {object} `this` for chaining
+    */
+   on: function(eventNames, listener, context) {
+      var eventNamesList;
+
+      if (typeof eventNames !== 'string') {
+         throw new Error('the eventNames parameter must be a string, but was: ' + (typeof eventNames));
+      }
+      if (typeof listener !== 'function') {
+         throw new Error('the listener parameter must be a function, but was: ' + (typeof listener));
+      }
+      eventNamesList = eventNames.split(' ');
+
+      // Remove the event listeners if they already exist. Listeners bound with this `on`
+      // function should always override listeners bound with `once`.
+      eventNamesList.forEach(function(eventName) {
```

Comment:
We've traditionally used underscore specifically so we can avoid the use of JS native array `forEach` and similar functions because of [browser support](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach#Browser_compatibility). Nowadays that basically means just IE8 would not be supported, which isn't that big of a deal. I do think, though, that we should likely note this in our README similar to how we made mention of the use of native promises.